src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotConstantPool.java

```diff
@@ -735,15 +717,7 @@ private static JavaType getJavaType(final Object type) {
 
     @Override
     public JavaMethod lookupMethod(int rawIndex, int opcode, ResolvedJavaMethod caller) {
-        int which; // interpretation depends on opcode
-        if (opcode == Bytecodes.INVOKEDYNAMIC) {
-            if (!isInvokedynamicIndex(rawIndex)) {
-                throw new IllegalArgumentException("expected a raw index for INVOKEDYNAMIC but got " + rawIndex);
-            }
-            which = rawIndex;
-        } else {
-            which = rawIndex;
-        }
+        int which = rawIndex;
```

Comment:
We could get rid of that intermediate variable now and just use `rawIndex` below.